pylint/checkers/classes/class_checker.py

```diff
@@ -1482,8 +1482,12 @@ def _check_invalid_class_object(self, node: nodes.AssignAttr) -> None:
         if not node.attrname == "__class__":
             return
         inferred = safe_infer(node.parent.value)
-        if isinstance(inferred, nodes.ClassDef) or inferred is astroid.Uninferable:
-            # If is uninferrable, we allow it to prevent false positives
+        if (
+            isinstance(inferred, nodes.ClassDef)
+            or inferred is astroid.Uninferable
+            or inferred is None
```

Comment:
```suggestion
            or inferred in {astroid.Uninferable, None}
```

Reply:
Happy to change, but it will be a bit slower, unless we make it a frozenset outside the condition. Do you think it's more readable on one line?

Reply:
It's just a suggestion for readability, I did not think the performance would be worse. It's probably not worth creating a constant for.